### ISSUE TITLE
Poor VGG performance in slim.

### ISSUE THREAD

**Issue by warmspringwinds** [tensorflow/tensorflow#5228]
Hello,

I have been following the slim tutorial:
https://github.com/tensorflow/models/blob/master/slim/slim_walkthough.ipynb

And decided to replace the Inception model v1 that is used in the example with VGG model
from tensorflow. I took it from here:
https://github.com/tensorflow/models/tree/master/slim#pre-trained-models

I have changed to code to work with VGG, but I got a considerably worse accuracy
even on simple images for VGG, while it works perfectly for Inception v1. Even though the
reported accuracy for VGG and Inception v1 are more or less equal in the imagenet.

I don't know if there is a bug in my code or the model is bad.

Here is the code that I have used:

```
%matplotlib inline

from matplotlib import pyplot as plt

import numpy as np
import os
import tensorflow as tf
import urllib2

from datasets import imagenet
from nets import vgg
from preprocessing import vgg_preprocessing

slim = tf.contrib.slim

image_size = vgg.vgg_19.default_image_size

checkpoints_dir = '/tmp/checkpoints'



with tf.Graph().as_default():
    url = 'https://upload.wikimedia.org/wikipedia/commons/7/70/EnglishCockerSpaniel_simon.jpg'
    url = 'https://upload.wikimedia.org/wikipedia/commons/d/d9/First_Student_IC_school_bus_202076.jpg'
    image_string = urllib2.urlopen(url).read()
    image = tf.image.decode_jpeg(image_string, channels=3)
    processed_image = vgg_preprocessing.preprocess_image(image, image_size, image_size, is_training=False)
    processed_images  = tf.expand_dims(processed_image, 0)
    
    # Create the model, use the default arg scope to configure the batch norm parameters.
    with slim.arg_scope(vgg.vgg_arg_scope()):
        logits, _ = vgg.vgg_19(processed_images, num_classes=1000, is_training=False)
    probabilities = tf.nn.softmax(logits)
    
    init_fn = slim.assign_from_checkpoint_fn(
        os.path.join(checkpoints_dir, 'vgg_19.ckpt'),
        slim.get_model_variables('vgg_19'))
    
    with tf.Session() as sess:
        init_fn(sess)
        np_image, probabilities = sess.run([image, probabilities])
        probabilities = probabilities[0, 0:]
        sorted_inds = [i[0] for i in sorted(enumerate(-probabilities), key=lambda x:x[1])]
        
    plt.figure()
    plt.imshow(np_image.astype(np.uint8))
    plt.axis('off')
    plt.show()
    

    names = imagenet.create_readable_names_for_imagenet_labels()
    for i in range(5):
        index = sorted_inds[i]
        print('Probability %0.2f%% => [%s]' % (probabilities[index], names[index]))
        
```

**Comment by warmspringwinds** [comment]
The issue was because for Resnet and VGG models the classes are shifted by 1.
So, inception has 1000 classes and one more for background while Resnet and VGG don't,
so the shift is needed.

More information can be found here: https://github.com/tensorflow/models/tree/master/slim#pre-trained-models

I think it makes sense to update the ipython notebook with a specific example for Resnet or VGG.
